### DISCUSSION TITLE
[Question]: How to remove country code from URL and all the related logic

### DISCUSSION THREAD

**Discussion by of0o** [medusajs/nextjs-starter-medusa#395] [Q&A]
Had been trying everything but no luck so far. I tried to move everything outside [countryCode] folder, and also adjusted my logic in middleware.ts. It worked for homepage but when I navigated to my product page, no items displayed. Is there a centralized place that we can turn this off, so it feels like the entire front-end codebase doesn't have countryCode in the logic? it would be very painful if we have to search for individual code and turn it off individually, and i'm a little surprised there's no answers for this after searching on a few forums/channels. I thought this is a common need for digital products.

**Comment by VariableVic** [comment]
Hey @of0o,

Good question, and sorry for the late reply! I can see why it would be helpful to kill the `[countryCode]` param if you only operate in one single region.

There is no magic switch to turn off the multi-region setup. However, the `countryCode` is mostly used by the `getRegion` function in `src/lib/data/regions.ts`.

You could refactor `getRegion` to no longer require the `countryCode` param and always return the region of your choice. Then, a simple find and replace `getRegion(countryCode)` -> `getRegion()` throughout the codebase should do the trick.

You'd also need to kill the url rewrite logic in `middleware.ts`.

This is from the top of my head, and there's probably be a few edge cases not covered by this. But it should at least point you in the right direction.

**Comment by AKRking** [comment]
this is the most rudimentary feature that should be come out of the box. very stupid not to add this to the settings
